File: app/src/main/java/com/example/healthtracker/feature/tmb/Tmb.java
package com.example.healthtracker.feature.tmb;

import com.example.healthtracker.common.base.BasePresenter;
import com.example.healthtracker.common.base.BaseView;
import com.example.healthtracker.model.CalcDao;

public interface Tmb {

    interface View extends BaseView {
        void onRegisterTmbValue();
    }

    interface Presenter extends BasePresenter {
        void registerTmbValue(double tmb, CalcDao dao);
        boolean validate(String height, String weight, String age);
        double calculateTmb(boolean isMan, int height, int weight, int age);
        double tmbAdaptedForLifestyle(String currentLifeStyle, double tmb, String[] arrayLifestyle);
    }
}

File: app/src/main/java/com/example/healthtracker/feature/tmb/Tmb.kt
package com.example.healthtracker.feature.tmb

import com.example.healthtracker.model.CalcDao

interface Tmb {

    interface View{
        fun onRegisterTmbValue()
        fun displayFailure(message: String)
    }

    interface Presenter{
        fun registerTmbValue(tmb: Double, type: String, dao: CalcDao)
        fun validate(height: String, weight: String, age: String): Boolean
        fun calculateTmb(isMan: Boolean, height: Int, weight: Int, age: Int): Double
        fun tmbAdaptedForLifestyle(currentLifeStyle: String, tmb: Double, arrayLifestyle: Array<String>): Double
    }

}